upup/pkg/fi/cloudup/apply_cluster.go

```diff
@@ -1055,21 +1055,42 @@ func (c *ApplyClusterCmd) addFileAssets(assetBuilder *assets.AssetBuilder) error
 		}
 
 		kubernetesVersion, _ := util.ParseKubernetesVersion(c.Cluster.Spec.KubernetesVersion)
-		if apiModel.UseExternalECRCredentialsProvider(*kubernetesVersion, c.Cluster.Spec.GetCloudProvider()) {
-			binaryLocation := c.Cluster.Spec.CloudProvider.AWS.BinariesLocation
-			if binaryLocation == nil {
-				binaryLocation = fi.PtrTo("https://artifacts.k8s.io/binaries/cloud-provider-aws/v1.27.1")
-			}
 
-			k, err := url.Parse(fmt.Sprintf("%s/linux/%s/ecr-credential-provider-linux-%s", *binaryLocation, arch, arch))
-			if err != nil {
-				return err
-			}
-			u, hash, err := assetBuilder.RemapFileAndSHA(k)
-			if err != nil {
-				return err
+		if ok, cloudProvider := apiModel.UseExternalKubeletCredentialProvider(*kubernetesVersion, c.Cluster.Spec.GetCloudProvider()); ok {
+			switch cloudProvider {
+			case kops.CloudProviderGCE:
+				binaryLocation := c.Cluster.Spec.CloudProvider.GCE.BinariesLocation
+				if binaryLocation == nil {
+					binaryLocation = fi.PtrTo("https://storage.googleapis.com/k8s-staging-cloud-provider-gcp/auth-provider-gcp")
```

Comment:
I think cloud-provider-gcp should cut a release before merging.

Reply:
yes, that should be pretty easy, I guess

Reply:
actually, it wan't pinned until a few days ago, so we should be good with a reminder issue, tagged with blocks-next.